src/ToricVarieties/NormalToricVarieties/constructors.jl

```diff
@@ -347,7 +350,13 @@ function weighted_projective_space(::Type{NormalToricVariety}, w::Vector{T}; set
     tr,_ = pseudo_inv(lattice_gens)
     ray_gens = ray_gens * transpose(tr)
     mc = IncidenceMatrix(subsets(Vector{Int}(1:length(w)), length(w)-1))
-    variety = NormalToricVariety(PolyhedralFan(ray_gens, mc; non_redundant=true ))
+    variety = normal_toric_variety(PolyhedralFan(ray_gens, mc; non_redundant=true ))
+    
+    # make standard choice for the weights of the cox ring
+    set_attribute!(variety, :torusinvariant_weil_divisor_group, free_abelian_group(length(w)))
+    set_attribute!(variety, :class_group, free_abelian_group(1))
+    weights = matrix(ZZ,hcat([w[i] for i in 1:length(w)]))
```

Comment:
```suggestion
    weights = matrix(ZZ,hcat(w))
```
That's shorter *and* allocs less / is faster (not that this matter much here).

I think we could improve this in a bunch of places outside of your code, too (don't worry about that, though, I just wanted to point out that multiple people, including me until today, seem not to be aware of this...)

Reply:
Thx for the pointer. We should look into this eventually. I put it in a separate issue (https://github.com/oscar-system/Oscar.jl/issues/1969), so we do not forget about this.